docs/spec.md

```diff
@@ -176,6 +181,55 @@ Informs the client about parameters. Only supported if the server declares the `
 }
 ```
 
+### Advertise Services
+
+Informs the client about available services. Only supported if the server declares the `services` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"advertiseServices"`
+- `services`: array of:
+  - `id`: number. The server may reuse ids when services disappear and reappear, but only if the services keeps the exact same name, type, and schema. Clients will use this unique id to cache schema info and deserialization routines.
+  - `name`: string
```

Comment:
It may happen that a service is unadvertised and a new service with the same name but different type is advertised. In this case the name is not unique.